### PR TITLE
[SPARK-38685][SQL] Improve the implement of percentile_cont

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The current implement of percentile_cont looks a little overkill.
This PR uses tree node tag to avoid the issue.

### Why are the changes needed?
Improve the implement of percentile_cont

### Does this PR introduce _any_ user-facing change?
'No'.
Just update new feature

### How was this patch tested?
tests updated.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @MaxGekk cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 1]
```diff
@@ -226,9 +226,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             // Only allow window functions with an aggregate expression or an offset window
             // function or a Pandas window UDF.
             w.windowFunction match {
-              case AggregateExpression(_: PercentileCont, _, _, _, _)
-                if w.windowSpec.orderSpec.nonEmpty || w.windowSpec.frameSpecification !=
-                    SpecifiedWindowFrame(RowFrame, UnboundedPreceding, UnboundedFollowing) =>
+              case AggregateExpression(percentile: Percentile, _, _, _, _)
```
Comment (beliefer) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:229]: Does match Percentile looks overkill too? @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 2]
```diff
@@ -1836,11 +1836,12 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
...
     val sortOrder = visitSortItem(ctx.sortItem)
-    val percentileCont = sortOrder.direction match {
-      case Ascending => PercentileCont(sortOrder.child, percentage)
-      case Descending => PercentileCont(sortOrder.child, Subtract(Literal(1), percentage))
+    val percentile = sortOrder.direction match {
+      case Ascending => new Percentile(sortOrder.child, percentage)
+      case Descending => new Percentile(sortOrder.child, Subtract(Literal(1), percentage))
     }
-    val aggregateExpression = percentileCont.toAggregateExpression()
+    percentile.setTagValue(Percentile.IS_ORDERED_SET, true)
```
[the remark this answers is not in the document]
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: OK

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala [hunk 3]
```diff
@@ -88,7 +88,8 @@ case class Percentile(
     this(child, percentageExpression, frequency, 0, 0)
   }
 
-  override def prettyName: String = "percentile"
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala:91]: We should override `nodeName` here. The default implementation of `prettyName` can handle `FunctionRegistry.FUNC_ALIAS` already

File: sql/core/src/test/resources/sql-tests/results/group-by.sql.out [hunk 4]
```diff
@@ -978,7 +978,7 @@ SELECT
  percentile_cont(0.25) WITHIN GROUP (ORDER BY v DESC)
 FROM aggr
 -- !query schema
-struct<percentile_cont(v, 0.25):double,percentile_cont(v, (1 - 0.25)):double>
+struct<percentile(v, 0.25, 1):double,percentile(v, (1 - 0.25), 1):double>
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/group-by.sql.out:981]: we should not have this behavior change.

**Comment by beliefer** [comment]
Based on discussion offline between @cloud-fan and me, discard this change.

**Pull request closed without merging**